ghost/core/test/integration/services/mega.test.js

```diff
@@ -94,4 +101,119 @@ describe('MEGA', function () {
+
+            // Prepare a post and email model
+            const emailModel = await createPublishedPostEmail();
+
+            // Launch email job
+            await _sendEmailJob({emailModel, options: {}});
+
+            await emailModel.refresh();
+            emailModel.get('status').should.eql('submitted');
+
+            // Get email data (first argument to sendStub call)
+            const emailData = sendStub.args[0][0];
+            const recipientData = sendStub.args[0][1];
+            const replacements = sendStub.args[0][2];
+
+            const recipient = recipientData[Object.keys(recipientData)[0]];
+
+            // Do the actual replacements for the first member, so we don't have to worry about them anymore
+            replacements.forEach((replacement) => {
+                emailData[replacement.format] = emailData[replacement.format].replace(
+                    replacement.match,
+                    recipient[replacement.id]
+                );
+
+                // Also force Mailgun format
+                emailData[replacement.format] = emailData[replacement.format].replace(
+                    `%recipient.${replacement.id}%`,
+                    recipient[replacement.id]
+                );
+            });
+
+            const html = emailData.html;
+
+            const memberUuid = recipient.unique_id;
+
+            // Test if all links are replaced and contain the member id
+            const cheerio = require('cheerio');
+            const $ = cheerio.load(html);
+
+            const exclude = '%recipient.unsubscribe_url%';
+            let firstLink;
+    
+            for (const el of $('a').toArray()) {
+                const href = $(el).attr('href');
+
+                if (href === exclude) {
+                    continue;
+                }
+
+                // Check if the link is a tracked link
+                assert(href.includes('?m=' + memberUuid), href + ' is not tracked');
+
+                if (!firstLink) {
+                    firstLink = new URL(href);
+                }
+            }
+
+            const links = await linkRedirectRepository.getAll({post_id: emailModel.get('post_id')});
+            const link = links.find(l => l.from.pathname === firstLink.pathname);
+            assert(link, 'Link model not created');
+
+            // Mimic a click on a link
+            const path = firstLink.pathname + firstLink.search;
+            await frontendAgent.get(path)
+                .expect('Location', link.to.href)
+                .expect(302);
+
+            // Since this is all event based, we need to wait a coulple of ms
+            // in the future we should wait for all dispatched events to be completed.
+            await sleep(200);
+
+            // Check if click was tracked and associated with the correct member
+            const member = await models.Member.findOne({uuid: memberUuid});
+            const clickEvent = await models.MemberLinkClickEvent.findOne({
```

Comment:
Sorry I missed this one, but same here!!

Reply:
Sorry, also missed this one! Only sad thing is that we cannot search on uuid directly (now we search member on uuid, to convert that id back to uuid later in the repository)

Reply:
Yeah it's a real shame - this is one of the downsides of having two unique identifiers - we can look at optimising it in future